server/channel.js

```diff
@@ -0,0 +1,72 @@
+var WebSocketServer = require("ws").Server;
+var fs              = require("fs");
+
+// this channel.js is the server-side abstraction over websockets
```

Comment:
Alright, turned out better than I had expected. I've refactored it to use prototype OOP as well.
You can view it [here](http://pastebin.com/duTreUcc).